File: backend/models/HealthDeclaration.js
import mongoose from 'mongoose';

const healthDeclarationSchema = new mongoose.Schema({
  donor: { type: mongoose.Schema.Types.ObjectId, ref: 'Donor', required: true },
  appointment: { type: mongoose.Schema.Types.ObjectId, ref: 'DonationAppointment', required: true },
  qrCode: { type: String }, // data URL của QR
  answers: {
    usedStimulants: { type: Boolean, default: false },
    usedCannabis: { type: Boolean, default: false },
    usedCocaine: { type: Boolean, default: false },
    usedHeroin: { type: Boolean, default: false },
    usedCorticoids: { type: Boolean, default: false },
    hasFever: { type: Boolean, default: false },
    hasCough: { type: Boolean, default: false },
    hasInfection: { type: Boolean, default: false },
    hasSkinDisease: { type: Boolean, default: false },
    hasAllergy: { type: Boolean, default: false }
  },
  createdAt: { type: Date, default: Date.now },
  expiresAt: { type: Date, default: () => new Date(Date.now() + 30 * 60 * 1000) } // 30 phút
});

export default mongoose.model('HealthDeclaration', healthDeclarationSchema);

File: backend-go/internal/models/health_declaration.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type HealthDeclaration struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	Donor       primitive.ObjectID `bson:"donor" json:"donor"`
	Appointment primitive.ObjectID `bson:"appointment" json:"appointment"`
	QRCode      string             `bson:"qrCode,omitempty" json:"qrCode,omitempty"`
	Answers     struct {
		UsedStimulants bool `bson:"usedStimulants" json:"usedStimulants"`
		UsedCannabis   bool `bson:"usedCannabis" json:"usedCannabis"`
		UsedCocaine    bool `bson:"usedCocaine" json:"usedCocaine"`
		UsedHeroin     bool `bson:"usedHeroin" json:"usedHeroin"`
		UsedCorticoids bool `bson:"usedCorticoids" json:"usedCorticoids"`
		HasFever       bool `bson:"hasFever" json:"hasFever"`
		HasCough       bool `bson:"hasCough" json:"hasCough"`
		HasInfection   bool `bson:"hasInfection" json:"hasInfection"`
		HasSkinDisease bool `bson:"hasSkinDisease" json:"hasSkinDisease"`
		HasAllergy     bool `bson:"hasAllergy" json:"hasAllergy"`
	} `bson:"answers" json:"answers"`
	CreatedAt time.Time `bson:"createdAt" json:"createdAt"`
	ExpiresAt time.Time `bson:"expiresAt" json:"expiresAt"`
}
